browser/brave_wallet/android/asset_ratio_controller_factory_android.cc

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/android/jni_android.h"
+#include "brave/browser/brave_wallet/asset_ratio_controller_factory.h"
+#include "brave/build/android/jni_headers/AssetRatioControllerFactory_jni.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+
+namespace chrome {
+namespace android {
+static jint JNI_AssetRatioControllerFactory_GetInterfaceToAssetRatioController(
+    JNIEnv* env) {
+  auto* profile = ProfileManager::GetActiveUserProfile()->GetOriginalProfile();
```

Comment:
we should remove `GetOriginalProfile` here. I think this was a mistake carried over from the JNI code. We should always pass the actual profile (regular or incognito) and let the service factory decide what to do. We're circumventing that logic here with `GetOriginalProfile`